src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -243,9 +243,14 @@ public virtual void Write(char[] chars, int index, int count)
         //
         public virtual void Write(double value)
         {
-            Span<byte> buffer = stackalloc byte[sizeof(double)];
-            BinaryPrimitives.WriteDoubleLittleEndian(buffer, value);
-            OutStream.Write(buffer);
+            if (BitConverter.IsLittleEndian)
+            {
+                WriteBytes(value);
+            }
+            else
+            {
+                WriteBytes(BinaryPrimitives.ReverseEndianness(BitConverter.DoubleToUInt64Bits(value)));
+            }
```

Comment:
In the changes below it's written _inline_, can this be made consistent (in either way)?

Reply:
Here, we call `WriteBytes<Double>` or `WriteBytes<UInt64>`, so the _other_ methods would need to change, e.g.:

```csharp
    public virtual void Write(ulong value)
    {
        if (BitConverter.IsLittleEndian)
        {
            WriteBytes(value);
        }
        else
        {
            WriteBytes(BinaryPrimitives.ReverseEndianness(value));
        }
    }
```

Reply:
@gfoidl pushed [46ea39b](https://github.com/dotnet/runtime/pull/99775/commits/46ea39ba82b619bc395b0706ec8c38f06e3d8bac)